### ISSUE TITLE
server: Minio version14 can not access by amz s3 java sdk

### ISSUE THREAD

**Issue by wangtao9090** [minio/minio#4004]
## Expected Behavior

minio version13 can access by amz s3 java sdk，but minio version14 can not access by amz s3 java sdk

minio version14 can  access by amz s3 java sdk

## Current Behavior

com.amazonaws.services.s3.model.AmazonS3Exception: header does not match what was computed. (Service: Amazon S3; Status Code: 400; Error Code: XAmzContentSHA256Mismatch; Request ID: 3L137)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1389) ~[nts-worker.jar:na]

java.io.IOException: com.amazonaws.services.s3.model.AmazonS3Exception: " (Service: Amazon S3; Status Code: 400; Error Code: XMinioInvalidObjectName; Request ID: 3L137), S3 Extended Request ID: 3L137

java.io.IOException: com.amazonaws.services.s3.model.AmazonS3Exception: The Content-Md5 you specified did not match what we received. (Service: Amazon S3; Status Code: 400; Error Code: BadDigest; Request ID: 3L137), S3 Extended Request ID: 3L137

## Steps to Reproduce (for bugs)

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.BufferedReader;

import com.amazonaws.regions.Region;
import com.amazonaws.regions.Regions;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.services.s3.S3ClientOptions;
import com.amazonaws.auth.AWSCredentials;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3Client;
import com.amazonaws.ClientConfiguration;

import com.amazonaws.AmazonClientException;
import com.amazonaws.AmazonServiceException;
import com.amazonaws.auth.profile.ProfileCredentialsProvider;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3Client;
import com.amazonaws.services.s3.model.PutObjectRequest;
import com.amazonaws.services.s3.model.GetObjectRequest;
import com.amazonaws.services.s3.model.S3Object;
import static com.amazonaws.SDKGlobalConfiguration.DISABLE_CERT_CHECKING_SYSTEM_PROPERTY;

public class S3Sample {
    private static String bucketName     = "testbucket";
    private static String keyName        = "hosts";
    private static String uploadFileName = "/etc/hosts";

    public static void main(String[] args) throws IOException {
    AWSCredentials credentials = new BasicAWSCredentials("YOUR-ACCESSKEYID",
                                 "YOUR-SECRETACCESSKEY");
    ClientConfiguration clientConfiguration = new ClientConfiguration();
    clientConfiguration.setSignerOverride("AWSS3V4SignerType");
    AmazonS3 s3Client = new AmazonS3Client(credentials, clientConfiguration);
    Region usEast1 = Region.getRegion(Regions.US_EAST_1);
    s3Client.setRegion(usEast1);
    s3Client.setEndpoint("http://localhost:9000");
    final S3ClientOptions clientOptions = S3ClientOptions.builder().setPathStyleAccess(true).build();
    s3Client.setS3ClientOptions(clientOptions);

    try {
        System.out.println("Uploading a new object to S3 from a file\n");
        File file = new File(uploadFileName);
        // Upload file
        s3Client.putObject(new PutObjectRequest(bucketName, keyName, file));

        // Download file
        GetObjectRequest rangeObjectRequest = new GetObjectRequest(bucketName, keyName);
        S3Object objectPortion = s3Client.getObject(rangeObjectRequest);
        System.out.println("Printing bytes retrieved:");
        displayTextInputStream(objectPortion.getObjectContent());
    } catch (AmazonServiceException ase) {
        System.out.println("Caught an AmazonServiceException, which " +
                   "means your request made it " +
                                       "to Amazon S3, but was rejected with an error response" +
                   " for some reason.");
        System.out.println("Error Message:    " + ase.getMessage());
        System.out.println("HTTP Status Code: " + ase.getStatusCode());
        System.out.println("AWS Error Code:   " + ase.getErrorCode());
        System.out.println("Error Type:       " + ase.getErrorType());
        System.out.println("Request ID:       " + ase.getRequestId());

    } catch (AmazonClientException ace) {
        System.out.println("Caught an AmazonClientException, which " +
                   "means the client encountered " +
                                       "an internal error while trying to " +
                                       "communicate with S3, " +
                   "such as not being able to access the network.");
        System.out.println("Error Message: " + ace.getMessage());

    }

    }

    private static void displayTextInputStream(InputStream input)
    throws IOException {
    // Read one text line at a time and display.
    BufferedReader reader = new BufferedReader(new InputStreamReader(input));
    while (true) {
        String line = reader.readLine();
        if (line == null) break;

        System.out.println("    " + line);
    }
    System.out.println();       
    }
}

## Your Environment

* Version used:14
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version:CentOS7.1 64  dibian 7.11 64
* Link to your project:

**Comment by harshavardhana** [comment]
We fixed this in master branch, should be available in the next stable release.

```
commit b62cd8ed84eb29cab0b667e07e671f33c1465760
Author: Harshavardhana <harsha@minio.io>
Date:   Mon Mar 27 17:02:04 2017 -0700

    sign/streaming: Content-Encoding is not set in newer aws-java-sdks (#3986)
    
    We can't use Content-Encoding to verify if `aws-chunked` is set
    or not. Just use 'streaming' signature header instead.
    
    While this is considered mandatory, on the contrary aws-sdk-java
    doesn't set this value
    
    http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html
    
    ```
    Set the value to aws-chunked.
    ```
    
    We will relax it and behave appropriately. Also this PR supports
    saving custom encoding after trimming off the `aws-chunked`
    parameter.
    
    Fixes #3983
```

**Comment by harshavardhana** [comment]
Closing as duplicate of #3983
